src/PointwiseFunctions/AnalyticData/WaveEquation/RegularSphericalWaveKerrSchild.hpp

```diff
@@ -0,0 +1,173 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <memory>
+#include <utility>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticData/AnalyticData.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/WaveEquation/RegularSphericalWave.hpp"
+#include "PointwiseFunctions/GeneralRelativity/TagsDeclarations.hpp"
+#include "PointwiseFunctions/MathFunctions/MathFunction.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+namespace Tags {
+template <typename Tag>
+struct dt;
+}  // namespace Tags
+/// \endcond
+
+// IWYU pragma: no_include <pup.h>
+
+namespace CurvedScalarWave {
+namespace AnalyticData {
+/*!
+ * \brief A 3D spherical wave solution to the covariant wave equation with
+ * a Kerr black hole providing the spacetime background
+ *
+ * \details Here we obtain the flat-spacetime scalar-wave system's solution
+ * from `ScalarWave::Solutions::RegularSphericalWave` and suitably augment the
+ * auxiliary variable to match that of the curved scalar-wave system. We keep
+ * the scalar wave's profile, as a function of coordinates, the same.
+ */
+class RegularSphericalWaveKerrSchild : public MarkAsAnalyticData {
+ public:
+  static constexpr size_t volume_dim = 3;
+
+  struct WaveProfile {
+    using type = std::unique_ptr<MathFunction<1>>;
+    static constexpr OptionString help = {
+        "The radial profile of the spherical wave."};
+  };
+  struct BlackHoleMass {
+    using type = double;
+    static constexpr OptionString help = {"Mass of the black hole"};
+    static type default_value() noexcept { return 1.; }
+    static type lower_bound() noexcept { return 0.; }
+  };
+  struct BlackHoleSpin {
+    using type = std::array<double, volume_dim>;
+    static constexpr OptionString help = {
+        "The [x,y,z] dimensionless spin of the black hole"};
+    static type default_value() noexcept { return {{0., 0., 0.}}; }
+  };
+  struct BlackHoleCenter {
+    using type = std::array<double, volume_dim>;
+    static constexpr OptionString help = {
+        "The [x,y,z] center of the black hole"};
+    static type default_value() noexcept { return {{0., 0., 0.}}; }
+  };
+  using options =
+      tmpl::list<BlackHoleMass, BlackHoleSpin, BlackHoleCenter, WaveProfile>;
+  static constexpr OptionString help{
+      "A regular spherical wave solution of the covariant wave equation"
+      " with a Kerr black hole (in Kerr-Schild coordinates) as background.\n"
+      "WARNING: This solution should only be used to set initial data."};
```

Comment:
This is absolutely "data" and anything that suggests otherwise needs to be changed. I've clarified the phrasing here as well as have written out the source equations explicitly above.